packages/cdk-assets/lib/private/docker-credentials.ts

```diff
@@ -0,0 +1,83 @@
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import { IAws } from '../aws';
+import { Logger } from './shell';
+
+export interface DockerCredentials {
+  readonly Username: string;
+  readonly Secret: string;
+}
+
+export interface DockerCredentialsConfig {
+  readonly version: string;
+  readonly domainCredentials: Record<string, DockerDomainCredentialSource>;
+}
+
+export interface DockerDomainCredentialSource {
+  readonly secretsManagerSecretId?: string;
+  readonly ecrRepository?: boolean;
+  readonly assumeRoleArn?: string;
+}
+
+/** Returns the presumed location of the CDK Docker credentials config file */
+export function cdkCredentialsConfigFile(): string {
+  return process.env.CDK_DOCKER_CREDS_FILE ?? path.join((os.userInfo().homedir ?? os.homedir()).trim() || '/', '.cdk', 'cdk-docker-creds.json');
+}
+
+let _cdkCredentials: DockerCredentialsConfig | undefined;
+/** Loads and parses the CDK Docker credentials configuration, if it exists. */
+export function cdkCredentialsConfig(): DockerCredentialsConfig | undefined {
+  if (!_cdkCredentials) {
+    try {
+      _cdkCredentials = JSON.parse(fs.readFileSync(cdkCredentialsConfigFile(), { encoding: 'utf-8' })) as DockerCredentialsConfig;
+    } catch (err) { }
+  }
+  return _cdkCredentials;
+}
+
+/** Fetches login credentials from the configured source (e.g., SecretsManager, ECR) */
+export async function fetchDockerLoginCredentials(aws: IAws, config: DockerCredentialsConfig, domain: string) {
+  if (!Object.keys(config.domainCredentials).includes(domain)) {
+    throw new Error(`unknown domain ${domain}`);
+  }
+
+  const domainConfig = config.domainCredentials[domain];
+
+  if (domainConfig.secretsManagerSecretId) {
+    const sm = await aws.secretsManagerClient({ assumeRoleArn: domainConfig.assumeRoleArn });
+    const secretValue = await sm.getSecretValue({ SecretId: domainConfig.secretsManagerSecretId }).promise();
+    if (!secretValue.SecretString) { throw new Error(`unable to fetch SecretString from secret: ${domainConfig.secretsManagerSecretId}`); };
+
+    const secret = JSON.parse(secretValue.SecretString);
+    if (!secret.username || !secret.secret) {
```

Comment:
Any plans on making these configurable? Might be worthwhile for added flexibility...

Reply:
Certainly considered it. Decided to KISS to start; I think it will be reasonably straightforward to add later as a follow-up.

Reply:
I would kindly ask you to add it in. Yes it's straightforward to add later on but I can nearly guarantee someone's going to ask for it right away, and we'll just put it on the backlog then. Might as well add it while the cement is still fresh?